scenes/test_array.txt

```diff
@@ -241,6 +241,239 @@
 *temp test_arr_max test_arr_max_store
 *gosub test_finish "Bug: cslib_array.filter_by_routine: Target array 'test_arr' of max size 3 is too small to hold all filtered values" cslib_bug_message 
 
+*comment :::::: REVERSE tests ::::::
+*gosub test_start "reverse #1 (even)"
+*set test_arr_1 1
+*set test_arr_2 2
+*set test_arr_3 3
+*set test_arr_4 4
+*set test_arr_5 5
+*set test_arr_6 6
+*set test_arr_count 6
+*gosub_scene cslib_array reverse "test_arr"
+*gosub test_assert_equal 6 test_arr_1
+*gosub test_assert_equal 5 test_arr_2
+*gosub test_assert_equal 4 test_arr_3
+*gosub test_assert_equal 3 test_arr_4
+*gosub test_assert_equal 2 test_arr_5
+*gosub test_assert_equal 1 test_arr_6
+*gosub test_finish
+
+*gosub test_start "reverse #2 (odd)"
+*set test_arr_1 1
+*set test_arr_2 2
+*set test_arr_3 3
+*set test_arr_count 3
+*gosub_scene cslib_array reverse "test_arr"
+*gosub test_assert_equal 3 test_arr_1
+*gosub test_assert_equal 2 test_arr_2
+*gosub test_assert_equal 1 test_arr_3
+*gosub test_finish
+
+*gosub test_start "reverse #3 (single)"
+*set test_arr_1 1
+*set test_arr_count 1
+*gosub_scene cslib_array reverse "test_arr"
+*gosub test_assert_equal 1 test_arr_1
+*gosub test_finish
+
+*gosub test_start "reverse #4 (empty)"
+*set test_arr_1 1
+*set test_arr_count 0
+*gosub_scene cslib_array reverse "test_arr"
+*gosub test_finish 1 test_arr_1
+
+
+*comment :::::: TO_STRING tests ::::::
+*gosub test_start "to_string #1"
+*set test_arr_1 1
+*set test_arr_2 2
+*set test_arr_3 3
+*set test_arr_count 3
+*gosub_scene cslib_array to_string "test_arr"
+*gosub test_finish "123" cslib_ret
+
+*gosub test_start "to_string #2"
+*set test_arr_1 "Hello"
+*set test_arr_2 "World"
+*set test_arr_count 2
+*gosub_scene cslib_array to_string "test_arr" " "
+*gosub test_finish "Hello World" cslib_ret
+
+*gosub test_start "to_string #3 (empty)"
+*set test_arr_count 0
+*gosub_scene cslib_array to_string "test_arr" " "
+*gosub test_finish "" cslib_ret
+
+*gosub test_start "to_string #4 (single)"
+*set test_arr_1 "Vanilla"
+*set test_arr_count 1
+*gosub_scene cslib_array to_string "test_arr" "DONOTUSEME"
+*gosub test_finish "Vanilla" cslib_ret
+
+
+*comment :::::: FOR_EACH tests ::::::
+*gosub test_start "for_each #1"
+*set test_arr_1 "john"
+*set test_arr_2 "jane"
+*set test_arr_count 2
+*gosub_scene cslib_array for_each "test_arr" "test_array" "for_each_capitalize"
+*if (false)
```

Comment:
that's a nice trick!! You could probably even do something like

*if (false and "Define a function")
*if ("Define a function" == "")

This is better than what my compiler does now:

*goto skip_xxx
*label xxx
FUNCTION BODY
*label skip_xxx

Reply:
I've seen GoldenSilver use it in his game files. (I think)

Reply:
I'm not sure I follow with the *if(false and "Define a function") stuff, but I'm curious?